controller/cache/admission.go

```diff
@@ -773,22 +773,21 @@ func doesMapContain(key, value string, propMap map[string]string) bool {
 	return false
 }
 
-// does propKey=propValue matches kvMap(criteria)
-func doesPropMatchCrt(propKey, propValue string, kvMap map[string]string) bool {
-	if value, exist := kvMap[propKey]; exist {
-		// kvMap contains propKey
-		if value != "" {
-			// does kvMap contain propKey=propValue?
-			return share.EqualMatch(value, propValue)
+func doesPropMatchCrt(propKey, propValue string, kvMap map[string][]string) bool {
```

Comment:
Is this function correct?
Let's say:
propKey="DESCRIPTION"
propValue="iperfserver"
kvMap={"DESCRIPTION": {"abc", "iperfserver"}}
This function will return false.
But is it expected to return false or true in this case?

Reply:
I see.

This function is only used in the following two places:
https://github.com/alopez-suse/neuvector/blob/372f93db1f2aaeaed73a3321f8441449d0f084a4/controller/cache/admission.go#L825
https://github.com/alopez-suse/neuvector/blob/372f93db1f2aaeaed73a3321f8441449d0f084a4/controller/cache/admission.go#L908

My change mimics what the previous logic was. Previously there could only be one value per key, so each new value would override the last, and so it would have returned false in that case. However, while this isn't introducing any new issues, when considering that this function is meant to represent `CriteriaOpContainsOtherThan` I think returning `true` is the desired behavior in that case.

Essentially this function should only return `false` when the `kvMap` variable doesn't contain anything matching `propKey` or `propValue`.

Reply:
Implemented changes to have `CriteriaOpContainsOtherThan` behave as discussed, changed the helper function name to be a bit clearer as to what its doing, and included a relevant test case.